integr8ly/integreatly-operator
# MGDAPI-3324 Enhance RhoamTenant CRD by adding ProvisioningStatus and TenantUrl fields

# Issue link
https://issues.redhat.com/browse/MGDAPI-3324

# What
As part of the onboarding of RHOAM to the Developer Sandbox, a new CRD, `RhoamTenant`, was added to control the creation of tenants. This PR expands on that CRD by adding two new fields, `ProvisioningStatus` and `TenantUrl`. The `ProvisioningStatus` field keeps track of whether the tenant's corresponding 3scale account has been requested or is ready. The `TenantUrl` field stores the tenant's system-developer route once it has been created.

This PR also removes the `RhoamTenant` CRD's yaml file from `config/crd/bases` and addresses some typos in the `Makefile`.

# Verification steps
1. Create a new [OSD cluster](https://qaprodauth.cloud.redhat.com/openshift/create/osd)
2. Checkout this PR
3. Once the cluster from step 1 is ready, target it and install RHOAM Multitenant:
```
export USE_CLUSTER_STORAGE="true"
INSTALLATION_TYPE=multitenant-managed-api make cluster/prepare/local
sleep 3m && oc patch rhmi rhoam -n sandbox-rhoam-operator --patch '{"metadata": {"annotations": {"in_prow": "true" }}}' --type=merge &
INSTALLATION_TYPE=multitenant-managed-api make code/run
```
4. Wait for the installation to `complete`:
```
oc get rhmi rhoam -o json | jq .status.stage
```
5. Run script to set up testing IDP:
```
DEDICATED_ADMIN_PASSWORD=Password1 PASSWORD=Password1 ./scripts/setup-sso-idp.sh
```
6. Login to the OpenShift console using one of the newly created user's credentials, i.e. `customer-admin01` and `Password1`. This is done in order to create a User CR for that user.
8. Create a new RhoamTenant CR with the same name as the User CR from step 6 (in this case `customer-admin01`)
9. Once the newly created RhoamTenant CR has finished reconciling (this should take less than a minute), verify that the CR has the following fields set in its `status` block:
    - `lastError`: `''`
    - `provisioningStatus`: `3scale account ready`
    - `tenantUrl`: `customer-admin01.apps.{whatever-the-cluster-name-is}.s1.devshift.org`
